kubernetes-client/src/Kubernetes/Client/Auth/Basic.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+module Kubernetes.Client.Auth.Basic where
+
+import           Data.ByteString.Base64         ( encode )
+import           Data.Function                  ( (&) )
+import           Data.Monoid                    ( (<>) )
+import           Data.Text                      ( Text )
+import           Kubernetes.Client.Auth.Internal.Types
+import           Kubernetes.OpenAPI.Core
+import           Kubernetes.Client.KubeConfig
+
+import qualified Data.Text.Encoding            as T
+import qualified Lens.Micro                    as L
+
+
+data BasicAuth = BasicAuth { basicAuthUsername :: Text
+                           , basicAuthPassword :: Text
+                           }
+
+instance AuthMethod BasicAuth where
+  applyAuthMethod _ BasicAuth{..} req =
+    pure
+      $           req
+      `setHeader` toHeader ("authorization", "Basic " <> encodeBasicAuth)
+      &           L.set rAuthTypesL []
+    where
+      encodeBasicAuth = T.decodeUtf8 $ encode $ T.encodeUtf8 $ basicAuthUsername <> ":" <> basicAuthPassword
```

Comment:
This has to be base64 encoded (at least according to [RFC 7617](https://tools.ietf.org/html/rfc7617). It is better to ues [`applyBasicAuth`](https://hackage.haskell.org/package/http-client-0.7.1/docs/Network-HTTP-Client.html#v:applyBasicAuth) function from http-client instead of `setHeader`.

Reply:
> This has to be base64 encoded (at least according to [RFC 7617](https://tools.ietf.org/html/rfc7617). It is better to ues [`applyBasicAuth`](https://hackage.haskell.org/package/http-client-0.7.1/docs/Network-HTTP-Client.html#v:applyBasicAuth) function from http-client instead of `setHeader`.

@akshaymankar So the `encode` function is from `Data.ByteString.Base64` so this encoding-decoding stuff what it does is
Text -> ByteString -> Encode to base 64 -> Text. I understand it is quite ugly but it is encoding the username and password as base64.

I opted for converting to `ByteString` then encode, and then convert to `Text` again because the library base64-bytestring is already imported and did not want to introduce another library for just doing the base64 encoding.

Will do the test.

Reply:
The `applyBasicAuth` works for raw `WAI` Requests, but here we are having `KubernetesRequest` which come from the OpenAPI generated code. I am not sure how I can use the `applyBasicAuth` here.

Reply:
You're right. I totally missed `encode` in all of this. I think this ok then :+1: